client/app/components/movies-connector/movieConnectorCache.factory.js

```diff
@@ -0,0 +1,73 @@
+(function() {
+    'use strict';
+    angular
+        .module('movies.connector')
+        .factory('moviesConnectorCache', moviesConnectorCacheFactory);
+
+    /**
+     * @ngInject
+     */
+    moviesConnectorCacheFactory.$inject = [
+        'cache_calls'
+    ];
+
+    function moviesConnectorCacheFactory(
+        cache_calls
+    ) {
+        var service = {
+            catchedInfo: [],
+            catchedSearch : [],
+            getCacheInfo: getCacheInfo,
+            addCacheInfo: addCacheInfo,
+            getCacheSearch: getCacheSearch,
+            addCacheSearch: addCacheSearch
+        };
+
+        function getCacheInfo(movieId) {
+            var result = {};
+            result.hasCache = false;
+            result.data = {};
+            angular.forEach(service.catchedInfo, function(obj){
```

Comment:
You could use an object with the movie id as keys, and then access directly without iteration